packages/parser/test/parser/parse.test.ts

```
@@ -1744,4 +1746,66 @@ multiline
+        const value = '"value"'
+        expect(parseValue(value).value).toEqual('value')
+      })
+    })
+    describe('when value is number', () => {
+      it('should return number value', () => {
+        const value = '1'
+        expect(parseValue(value).value).toEqual(1)
+      })
+    })
+    describe('when value is boolean', () => {
+      it('should return boolean value', () => {
+        const value = 'true'
+        expect(parseValue(value).value).toEqual(true)
+      })
+    })
+    describe('when value is reference expression', () => {
+      const value = new ReferenceExpression(new ElemID('adapter', 'someType', 'instance', 'someInstance'))
+      it('should return reference expression value', () => {
+        expect(parseValue('adapter.someType.instance.someInstance').value).toEqual(value)
+      })
+    })
+    describe('when consumeValue adds errors to context', () => {
+      it('should return the errors in the result', () => {
+        const mockError: ParseError = { message: 'Test error', detailedMessage: 'Test error' } as ParseError
+        const mockResult = { value: 'mockValue', range: {} as Omit<SourceRange, 'filename'> }
+
+        mockConsumeValue.mockImplementation(context => {
+          context.errors.push(mockError)
+          return mockResult
+        })
+
+        const result = parseValue('mock input')
+
+        expect(result.errors).toEqual([mockError])
+        expect(result.value).toEqual(mockResult.value)
+      })
+    })
+    describe('when consumeValue adds promises to context', () => {
+      it('should return the errors in the result', () => {
+        const mockResult = { value: 'mockValue', range: {} as Omit<SourceRange, 'filename'> }
+        mockConsumeValue.mockImplementation(context => {
+          context.valuePromiseWatchers.push({ parent: 'some value', key: 'some key' })
+          return mockResult
+        })
+        expect(() => parseValue('mock input')).toThrow('Unexpected promise')
```

the test says it should return the errors in the result (which would be the correct behavior in our APIs), but this actually asserts that the code throws an error instead of returning it